app/controllers/users_controller.rb

```diff
@@ -96,13 +96,23 @@ def create
     @user.password = temp_pass
     @user.password_confirmation = temp_pass
     @user.skip_confirmation!
-
-    if @user.save
+    save_worked = false
+    begin
+      save_worked = @user.save
+    rescue => error
+      error_message = error.message
+      if error_message.include? "Duplicate entry" and error_message.include? "@"
+        flash[:error] = "Email #{@user.email} already in use"
+      else
+        flash[:error] = error_message
+      end
+    end
+    if save_worked
       @user.send_reset_password_instructions
       flash[:success] = "User creation success"
       redirect_to(users_path) && return
     else
-      flash[:error] = "User creation failed"
+      #flash[:error] = "User creation failed"
```

Comment:
Why is this commented out? Also, let's change this to "Sorry, could not create user. Please try again later."